lib/elixir/lib/inspect.ex

```diff
@@ -39,21 +39,39 @@ defprotocol Inspect do
   other string `">"`.
 
   Since regular strings are valid entities in an algebra document,
-  an implementation of inspect may simply return a string,
-  although that will devoid it of any pretty-printing.
+  an implementation of the `Inspect` protocol may simply return a
+  string, although that will devoid it of any pretty-printing.
 
   ## Error handling
 
   In case there is an error while your structure is being inspected,
   Elixir will raise an `ArgumentError` error and will automatically fall back
   to a raw representation for printing the structure.
 
-  You can however access the underlying error by invoking the Inspect
-  implementation directly. For example, to test Inspect.MapSet above,
+  You can however access the underlying error by invoking the `Inspect`
+  implementation directly. For example, to test `Inspect.MapSet` above,
   you can invoke it as:
 
       Inspect.MapSet.inspect(MapSet.new(), %Inspect.Opts{})
 
+  ## Deriving
+
+  The `Inspect` protocol can be derived to hide certain fields from
+  structs, so they don't show up in logs, inspects and similar. This
+  is especially useful for fields containing private information.
+
+  The options `:only` and `:except` can be used with `@derive` to
+  specify which fields should and should not appear in the
+  algebra document:
+
+      defmodule User do
+        @derive {Inspect, only: [:id, :name]}
+        defstruct [:id, :name, :address]
+      end
+
+      inspect(%User{id: 1, name: "Homer", address: "742 Evergreen Terrace"})
+      #=> %User{id: 1, name: "Homer"}
```

Comment:
I think this can be misleading as there is no indication there are extra fields. So I propose two changes:

* Add `, ...` at the end
* Because adding `, ...` no longer makes it a pure Elixir data structure, we need to use `#User<...>`

Thoughts?

Reply:
The `User` struct only has three fields, but only `:id` and `:name` are included when inspecting :)

The given examples are actual results:

```elixir
$ bin/iex
Erlang/OTP 21 [erts-10.0.4] [source] [64-bit] [smp:8:8] [ds:8:8:10] [async-threads:1] [hipe] [dtrace]

Interactive Elixir (1.8.0-dev) - press Ctrl+C to exit (type h() ENTER for help)
iex(1)> defmodule User do
...(1)>   @derive {Inspect, only: [:id, :name]}
...(1)>   defstruct [:id, :name, :address]
...(1)> end
{:module, User,
 <<70, 79, 82, 49, 0, 0, 5, 164, 66, 69, 65, 77, 65, 116, 85, 56, 0, 0, 0, 181,
   0, 0, 0, 18, 11, 69, 108, 105, 120, 105, 114, 46, 85, 115, 101, 114, 8, 95,
   95, 105, 110, 102, 111, 95, 95, 7, 99, ...>>, %User{id: nil, name: nil}}
iex(2)> inspect(%User{id: 1, name: "Homer", address: "742 Evergreen Terrace"})
"%User{id: 1, name: \"Homer\"}"
iex(3)>
```

Reply:
Oh! I thought you were talking about the documentation, and not about the actual implementation. I misunderstood the requirements then.

And yes, it makes more sense to add `, ...` and use `#` 👍